core/src/main/java/org/elasticsearch/common/cli/CliToolConfig.java

```diff
@@ -189,8 +198,17 @@ public Builder optionGroups(OptionGroupBuilder... optionGroupBuilders) {
                 return this;
             }
 
+            /**
+             * Allow for lenient parsing. In case of unknown parameters, keep on parsing and put them into args instead
+             * of throwing an exception
+             */
+            public Builder lenient(boolean lenient) {
+                this.stopAtNonOption = lenient;
```

Comment:
Should `lenient` not be inverted. If `lenient` is `true` I would expect it to not stop at a non option so `this.stopAtNonOption` should be `false`?